internal/component/loki/source/file/tailer.go

```diff
@@ -180,26 +139,83 @@ func getLastLinePosition(path string) (int64, error) {
 	}
 }
 
+func (t *tailer) Run() {
+	fi, err := os.Stat(t.path)
+	if err != nil {
+		level.Error(t.logger).Log("msg", "failed to tail file", "path", t.path, "err", err)
+		return
+	}
+	pos, err := t.positions.Get(t.path, t.labelsStr)
+	if err != nil {
+		level.Error(t.logger).Log("msg", "failed to get file position", "err", err)
+		return
+	}
+
+	if fi.Size() < pos {
```

Comment:
If a file is rotated but the new rotated file is larger than the previous, do we miss the first `pos` lines?